### DISCUSSION TITLE
Rejected delta updates.

### DISCUSSION THREAD

**Discussion by nishaanth-ms-21155** [sparkle-project/Sparkle#2619] [Q&A] [answered]
I already use Sparkle v1.24 on my app and it works for whole update. Now to bring delta updates into existing app, I followed these steps. 
1. Used generate_keys which stored a key in Keychain and gave me a public key (eddsa)
2. Added the given key In Info.plist (Does it have to be in both new and old app? I added it in my latest .app only)
3. Generated .delta from new New.app and Old.app -  Old_New.delta
4. Used `sign_update Old_New.delta` which gave sparkle:edSignature, length I added to appcast.xml
5. Served My delta and added the URL
My appcast looks something like this.

```
<rss xmlns:dc="http://purl.org/dc/elements/1.1"
 xmlns:sparkle="http://www.andymatuschak.org/xml-namespaces/sparkle" version="2.0">
<channel>
<title>MyApp</title>
<link><--link-to-appcast.xml--></link>
<description>Desc</description>
<language>en</language>
<item>
<title>Title</title>
<description>Desc</description>
<pubDate>Date</pubDate>
<enclosure sparkle:shortVersionString="1.0.12.123" sparkle:version="12.123"
url="http://localhost:8000/Fake.dmg"/>
<sparkle:deltas>
<enclosure url="http://localhost:8000/Delta.delta"
type="application/octet-stream"
sparkle:version="12.123"
sparkle:shortVersionString="1.0.12.123"
sparkle:deltaFrom="12.120"
sparkle:edSignature="GeneratedSign=="
length="22233764"
/>
</sparkle:deltas>
</item>
</channel>
</rss>
```

Still delta is rejected and the Whole update happens. Am I missing something?

**Comment by zorgiepoo** [comment, marked as answer]
> Added the given key In Info.plist (Does it have to be in both new and old app? I added it in my latest .app only)

Yes it needs to be both old and new.

delta updates generated from 1.24 tools are also [too old](https://sparkle-project.org/documentation/delta-updates/). You will need to update (at least the tools) to 1.27.3 or preferably latest 2.x. 1.24 framework also has security issues so I don't recommend it. And also the delta updates generated in Sparkle 2 use a newer and better format (also check the [compatibility section](https://sparkle-project.org/documentation/delta-updates#compatibility) for generating deltas).
